CHANGELOG.md

```diff
@@ -1,3 +1,31 @@
+CHANGES IN VERSION 1.16.0 (FROM 1.15.1)
+===================================
+
+Appium 1.16.0 is a minor release
+
+#### General
+
+#### IOS (XCUITest):
+* Allow execution of WDA without app under test (https://github.com/appium/appium-xcuitest-driver/pull/1093)
+* Add `mobile: deleteFile` command for removing files/folders from the device (https://github.com/appium/appium-xcuitest-driver/pull/1095)
+* Fall back to `ios-deploy` if native install fails (https://github.com/appium/appium-xcuitest-driver/pull/1098)
+* Make sure ports are properly closed when cleaning up (https://github.com/appium/appium-xcuitest-driver/pull/1094)
+* Handle socket errors on session start (https://github.com/appium/appium-xcuitest-driver/pull/1101)
+* Properly handle creating simulators with patch versions (https://github.com/appium/node-simctl/pull/89)
+* Fix handling of Safari console log (https://github.com/appium/appium-remote-debugger/pull/176)
+* Add ability to interact with Shadow DOM in Safari (https://github.com/appium/appium-remote-debugger/commit/559395a16088142b27289dbac6d3a5ab36caa716)
+* New capabilities
+  * `safariLogAllCommunication` - log all plists passed to and received from the Web Inspector (https://github.com/appium/appium-xcuitest-driver/pull/1105)
+  * `safariLogAllCommunicationHexDump` - log the raw data passed to and received from the Web Inspector (https://github.com/appium/appium-xcuitest-driver/pull/1105)
+  * `safariSocketChunkSize` - change the size of the data passed to the Web Inspector on real devices (https://github.com/appium/appium-xcuitest-driver/pull/1105)
+  * `appPushTimeout` - timeout for application upload in millisecond, on real devices (https://github.com/appium/appium-xcuitest-driver/pull/1104)
+
+#### Android General
+#### Android (UiAutomator2)
+#### Android (Espresso)
```

Comment:
## General
- Add log a custom event and get events (https://github.com/appium/appium-base-driver/pull/364, https://github.com/appium/appium-base-driver/pull/365)
- Improve memory usage

## Espresso

- Allow configuring Gradle, AGP, various SDK and Kotlin versions for Espresso server (https://github.com/appium/appium-espresso-driver/pull/496)
    - Many thanks, @tinder-ktarasov (Maybe, we do not have documentation yet)
- Add get clipboard for Android 10+ (https://github.com/appium/appium-espresso-driver/pull/512)

## UIA2
- Add get clipboard for Android 10+ (hhttps://github.com/appium/appium-uiautomator2-driver/pull/348)
- Fix prefer IPv4 address to the host name (https://github.com/appium/appium-uiautomator2-driver/pull/343)
- Fix Turn write access verification on Windows environment (https://github.com/appium/appium-uiautomator2-driver/pull/340)
- Aff bluetooth state as deviceInfo (https://github.com/appium/appium-uiautomator2-server/pull/312)


## Windows
- Support WinAppDriver 1.2 release candidate (https://github.com/appium/appium-windows-driver/pull/60)
    - Important capabilities: `ms:waitForAppLaunch`, `ms:experimental-webdriver`
- feat: Add a possibility to customize application startup timeout (https://github.com/appium/appium-windows-driver/pull/59)
    - `createSessionTimeout`, https://github.com/appium/appium-windows-driver#windowsdriver-specific-capabilities